libvast/src/system/connector.cpp

```diff
@@ -0,0 +1,168 @@
+
+namespace vast::system {
+
+namespace {
+bool is_non_recoverable_error_enum(caf::sec err_enum) {
+  switch (err_enum) {
+    case caf::sec::none:
+    case caf::sec::unexpected_message:
+    case caf::sec::unexpected_response:
+    case caf::sec::request_receiver_down:
+    case caf::sec::request_timeout:
+    case caf::sec::no_such_group_module:
+    case caf::sec::no_actor_published_at_port:
+    case caf::sec::unexpected_actor_messaging_interface:
+    case caf::sec::state_not_serializable:
+    case caf::sec::unsupported_sys_key:
+    case caf::sec::unsupported_sys_message:
+    case caf::sec::disconnect_during_handshake:
+    case caf::sec::cannot_forward_to_invalid_actor:
+    case caf::sec::no_route_to_receiving_node:
+    case caf::sec::failed_to_assign_scribe_from_handle:
+    case caf::sec::failed_to_assign_doorman_from_handle:
+    case caf::sec::cannot_close_invalid_port:
+    case caf::sec::cannot_connect_to_node:
+    case caf::sec::cannot_open_port:
+    case caf::sec::network_syscall_failed:
+    case caf::sec::invalid_argument:
+    case caf::sec::invalid_protocol_family:
+    case caf::sec::cannot_publish_invalid_actor:
+    case caf::sec::cannot_spawn_actor_from_arguments:
+    case caf::sec::end_of_stream:
+    case caf::sec::no_context:
+    case caf::sec::unknown_type:
+    case caf::sec::no_proxy_registry:
+    case caf::sec::runtime_error:
+    case caf::sec::remote_linking_failed:
+    case caf::sec::cannot_add_upstream:
+    case caf::sec::upstream_already_exists:
+    case caf::sec::invalid_upstream:
+    case caf::sec::cannot_add_downstream:
+    case caf::sec::downstream_already_exists:
+    case caf::sec::invalid_downstream:
+    case caf::sec::no_downstream_stages_defined:
+    case caf::sec::stream_init_failed:
+    case caf::sec::invalid_stream_state:
+    case caf::sec::unhandled_stream_error:
+    case caf::sec::bad_function_call:
+    case caf::sec::feature_disabled:
+    case caf::sec::cannot_open_file:
+    case caf::sec::socket_invalid:
+    case caf::sec::socket_disconnected:
+    case caf::sec::socket_operation_failed:
+    case caf::sec::unavailable_or_would_block:
+    case caf::sec::malformed_basp_message:
+    case caf::sec::serializing_basp_payload_failed:
+    case caf::sec::redundant_connection:
+    case caf::sec::remote_lookup_failed:
+    case caf::sec::no_tracing_context:
+    case caf::sec::all_requests_failed:
+    case caf::sec::field_invariant_check_failed:
+    case caf::sec::field_value_synchronization_failed:
+    case caf::sec::invalid_field_type:
+    case caf::sec::unsafe_type:
+    case caf::sec::save_callback_failed:
+    case caf::sec::load_callback_failed:
+    case caf::sec::conversion_failed:
+    case caf::sec::connection_closed:
+    case caf::sec::type_clash:
+    case caf::sec::unsupported_operation:
+    case caf::sec::no_such_key:
+    case caf::sec::broken_promise:
+    case caf::sec::connection_timeout:
+    case caf::sec::action_reschedule_failed:
```

Comment:
Did anything speak against using `default:` here? Seems like you already select those that matter with `return true` below.

Reply:
I like the idea of having compilation error when the new caf::sec enum values are introduced. Such compilation error will force us to check if the newly introduced error code should also be considered as a non recoverable one.

Reply:
Good argument, I'm convinced.